docs/rfc/active/data-permissions/README.md

```diff
@@ -0,0 +1,138 @@
+- Druid
+- Elastic Search
+- Glue
+- Hive (via Apache Ranger)
+- Kafka
+- Looker
+- MariaDB
+- Metabase
+- MongoDB
+- MicrosoftSQLServer
+- MySQL
+- Oracle
+- PostgreSQL
+- Redash
+- Redshift
+- S3 (via IAM)
+- Sagemaker
+- Snowflake
+- Superset
+- Tableau
+- Trino
+
+## Requirements
+
+The permissions aspect must be:
+
+- Appliable to datasets, containers, dashboards
+- Support ingesting and viewing structured access control (GRANT {permission} ON {object} TO {user or role}
+- Support ingesting and viewing unstructured access control (IAM Policy Documents, Custom Policies from other datasets)
+-
+
+### Extensibility
+
+> Please also call out extensibility requirements. Is this proposal meant to be extended in the future? Are you adding
+> a new API or set of models that others can build on in later? Please list these concerns here as well.
+
+## Non-Requirements
+
+This RFC specifically calls for functionality to view the permissions. Actions taken on the permissions may be great areas for future enhancement but is not a focus here.
+
+## Detailed design
+
+There are no new Entities associated with this design.
+
+There is one aspect proposed:
+
+```
+record DataPermission {
+  structuredPermissions: array[StructuredDataPermission]
+  unstructuredPermissions: array[UnstructuredDataPermission]
+  relatedPermissions: array[Urn]
+}
+
+record StructuredDataPermission {
+   permission: string
+   object: Urn // if not specified, is implicitly the urn of the containing Entity
+   subject: union[corpUser, corpGroup, string] // grant to a user, group, or non-standard entity (e.g. "*", "all", "NONE")
+}
+
+record UnstructuredDataPermission {
+  language: string // maybe ENUM? (e.g. "IAM")
+  value: string
+}
+```
+
+This aspect should be attachable to
+
+- Datasets (e.g. SQL Grants on Tables)
+- Containers (e.g. Grants on Schemas/Databases)
+- Dashboards (e.g. Tableau permissions on dashboard access)
+- Charts (e.g. Tableau permissions on chart access)
+
+The `relatedPermissions` attribute allows us to expand Permissions into a full fledged Entity instead of just an aspect.
+
+The common case (SQL as an example) is primarily attached permissions (e.g. grants on a table or inline IAM policies).
+
+The extension allows for out of band policies (e.g. an external policy attached to multiple entities).
+
+## How we teach this
+
+We should ingest data from sources that support it automatically.
+Many sources already support permissions modeling and providing examples in user-guides will help people learn this.
+
+## Drawbacks
+
+Permissions on datasets can be restricted/sensitive data.
+As an aspect on many different entities, managing RBAC for the ability to view/query these permissions may be difficult.
+
+SQLAlchemy does not provide a standard mechanism via inspector for viewing permissions.
+Many SQL Sources support `information_schema` and thus we can query those tables to view these.
+This may end up with significant source-specific work to support.
```

Comment:
This is almost certainly true. Perhaps a permissions specific framework within the ingestion framework could be built to minimize the level of effort.